docs/dev-proc/hcl-maintainer.md

```diff
@@ -0,0 +1,31 @@
+# Dasharo HCL Maintainer documentation
+
+This documentaintion aims to describe how to maintain Dasharo Hardware
+Compatibility List for CPUs, memory, mainboards and GPU.
+
+## Generating entry
+
+### Dasharo HCL report parsing
+
+For reports uploaded to 3mdeb cloud please use [dedicated
+script](https://github.com/Dasharo/dts-hw-conf-gen) readme.
+
+## Extending list
+
+Please always sort table before publishing.
+
+In vim:
+
+- ++shift+v++ - to enter visual mode and mark whole range of entries.
+- type `:` and `sort` to run vim sort function on marked range.
+
+### Github report
+
+<!--
+Current template for adding HCL report over Github issue is way too complex.
+Nobody will wast time doing that.
```

Comment:
So how it should look like to not cause trouble for us? To not cause ping-pongs to get missing data? If somebody does not want to contribute directly, then should cope with some inconvenience of more complex reporting. It's for everybody's benefit, not one's convenience

Reply:
> So how it should look like to not cause trouble for us?

Let's think about community, users and customers first.

Tactics of complex HCL and coping with some inconvenience gives us no results. Number of HCL reports provided over GH issues template is zero. So we have clear evidence in number that it doesn't work.

Goal of HCL is to improve users life by providing information about compatibility. We sitting on ton of HCL reports (almost 3GB), most information from those reports are not exposed to users, all those reports were provided using DTS.

HCL reports have to be trivial.

We have to change it. I really believe linux-hardware.org and bsd-hardware.info is the way to go further.

Meanwhile we have to deal with still growing number of HCL reports which were sent to us and are of no use to community.

In case of HCL reports for debugging purposes this is different story and is out of scope of what I'm proposing here. DTS should have good enough dump to provide enough debugging information for diagnostics, also maybe hw-probe could would be good enough.

Reply:
> Tactics of complex HCL and coping with some inconvenience gives us no results. Number of HCL reports provided over GH issues template is zero. So we have clear evidence in number that it doesn't work.
> 
> Goal of HCL is to improve users life by providing information about compatibility. We sitting on ton of HCL reports (almost 3GB), most information from those reports are not exposed to users, all those reports were provided using DTS.
> 
> HCL reports have to be trivial.
> 
> We have to change it. I really believe linux-hardware.org and bsd-hardware.info is the way to go further.
> 
> Meanwhile we have to deal with still growing number of HCL reports which were sent to us and are of no use to community.

Well, I think community is learning slowly. As we kindly ask them for HCL reports each time they ask about issues, more and more HCLs are being attached to issues. See also this one: https://github.com/Dasharo/dasharo-issues/issues/409#issuecomment-1511217858

Aren't we jumping to conclusions too fast? And there I thought I am too fast with conclusions :)

Reply:
@miczyg1 let's try to go to a conclusion, what is your expectation in this conversation? If we can get some consensus with other reviewers on your expectations, I will apply those changes. I would like to move whole this topic further at some point.

@macpijan @sulewskiprzemyslaw for over two months, I would appreciate the feedback.

Reply:
@pietrushnic if we say we dislike something, we should a least also say how it should look like so that we will like it. Few questions that arise from this short compliant:

```
Current template for adding HCL report over Github issue is way too complex.
Nobody will wast time doing that.
```

1. What makes current template way too complex?
2. Maybe some sections of the templates are redundant and are making it overly complex? If yes which ones?
3. What are the criteria of non-complex template?

Reply:
> What makes current template way too complex?

You have to provide up to [20 data points (10 required, 10 optional)](https://github.com/Dasharo/dasharo-issues/blob/master/.github/ISSUE_TEMPLATE/hcl-repor,t.md)
- according to [this book](https:/ninewww.amazon.com/Checklist-Manifesto-How-Things-Right/dp/0312430000) most extended procedure for pilots has 9 steps because research shows that prolonged procedures lead to rapid growth in mistakes; IIRC on average procedures have 4-5 pints
- there is no explanation if it is an easy, automatic way of gathering information
- **nobody ever filled even one such report**
- format of providing that information is unknown;

> Maybe some sections of the templates are redundant and are making it overly complex? If yes which ones?

This is already explained in the section proposing a new look for tables. There is a very minimal amount of information needed and if one wants the link to the source is provided (e.g. [CPU](https://github.com/Dasharo/docs/pull/463/files#diff-475b421b7f7e689cac83c3872d82d0319ba246ba1d11737b6a1b676ed17bd861R43)).

IMHO there is no use for existing HCL reporting using GH template, this method exist for 9+ months and there no even a single HCL report provided.

> What are the criteria of non-complex template?

Complexity is not a key problem, an efficient solution is what we aim for. It is just a hypothesis about why we have a problem. We only aim for efficient HCL reporting and maintenance on our side. If complexity leads to better efficiency then let it be.

At minimum what is presented in the Checklist Manifesto book would be helpful—ideally fully automated reporting instead of manual reports on GH. 

To sum up, I want to achieve efficient HCL report gathering and processing. Right now, we need better efficiency, and the goal of this PR is to improve efficiency. If it solves the discussion, I will remove the HCL report template and dismiss claims about complexity.

Please express what are you expecting and if you know a better way to address a key problem that this PR tries to solve.

Reply:
> You have to provide up to [20 data points (10 required, 10 optional)](https://github.com/Dasharo/dasharo-issues/blob/master/.github/ISSUE_TEMPLATE/hcl-repor,t.md)

Ohh so you meant that one particular issue template, I see now what you meant.

> IMHO there is no use for existing HCL reporting using GH template, this method exist for 9+ months and there no even a single HCL report provided.

True. But we can only guess the reasons why there is not a single report yet added. Secondly, these are just templates that should help the community supply the information we need, they serve as a guidance, I have never treated it as fixed format. Although I agree, much more efficient approach would be good in some cases (e.g. drop a pack of logs instead of filling 10-20 options, however still certain data cannot be extracted from logs only - I can think of at least GPU exact model name, total VRAM, etc.).

> To sum up, I want to achieve efficient HCL report gathering and processing. Right now, we need better efficiency, and the goal of this PR is to improve efficiency. If it solves the discussion, I will remove the HCL report template and dismiss claims about complexity.

Well, I think we can remove it. But I still have some concerns about consuming the efficient HCL reports on our side. By consuming I mean extracting the value we need. Let's take two hypothetical users who submit HCL reports:

1. Regular user, who submits a bunch of logs and we have to extract value from it by ourselves.
2. Enthusiats user who experiments, sends us logs and screenshots, but also describe the results and summarizes it.

With type 1 we cannot expect the person to provide much more additional data even if we need it (most likely it will be to difficult for that person).

With type 2 there is high chance w can get almost any info we will eventually need. Furthermore type 2 reports have significant more value, much of the hard work is already done by the person and what is important we have incredible input for knowledge base and documentation.

We have a mass of type 1 reports on cloud pushed via DTS right? Are we doing something with them? Nope, almost nothing. It makes some use in extremely rare cases (returning the firmware dumps t their original owners excluded). In general, it also contains a configuration that is well tested by us.

While we have only a handful number of type 2 reports:
- https://github.com/Dasharo/dasharo-issues/issues/173
- another big one incoming from memory experiments from Dasharo IRC

They bring much more value, have much more sophisticated hardware configurations, reveal more bugs, often with clear reasons of their occurrence, and huge, invaluable amount of knowledge.

So my main question is now, where do we look for efficiency? On our side? or on the user side submitting such report?

Reply:
> Although I agree, much more efficient approach would be good in some cases (e.g. drop a pack of logs instead of filling 10-20 options, however still certain data cannot be extracted from logs only - I can think of at least GPU exact model name, total VRAM, etc.).

IMO if we cannot get data automatically we should not gather it. There is too much problem with data gathered that way.

> With type 2 there is high chance w can get almost any info we will eventually need. Furthermore type 2 reports have significant more value, much of the hard work is already done by the person and what is important we have incredible input for knowledge base and documentation.

I'm not sure how this comment affect whole HCL report efficiency topic.

> We have a mass of type 1 reports on cloud pushed via DTS right? Are we doing something with them? Nope, almost nothing. It makes some use in extremely rare cases (returning the firmware dumps t their original owners excluded). In general, it also contains a configuration that is well tested by us.

@miczyg1 this PR (and related work) is about that. There are also internal tasks regarding HCL (DTS improvements, Dasharo Transparent Validation improvements) which coming from this PR. This PR is the place to change way we treat HCL reports ideally make it automatic and close to zero manual effort. What I propose here are scripts which automatically generate tables based on HCL reports - those table can be copy pasted to Dasharo Universe documentation. This is very far from ideal but without discussion and further work on the topic not much will change.

> They bring much more value, have much more sophisticated hardware configurations, reveal more bugs, often with clear reasons of their occurrence, and huge, invaluable amount of knowledge.

Great. I'm not sure how it contradicts anything I propose in PR. It is even better we probably can use gh cli to pull files and extract value from packages to feed Dasharo Universe HCL pages.

> So my main question is now, where do we look for efficiency? On our side? or on the user side submitting such report?

Both, whole process is flawed. HCL report from DTS works pretty well what we can see by number of packages on our cloud. Key issue is that we have value sitting there which we do not process. More to that SEO shows Dasharo Universe benefit from compatibility lists because many people look for such information. TBH we should have HCL reports for all supported hardware.

Reply:
@miczyg1 ping

Reply:
@miczyg1 ping2